Pwx 44553 rhel9.6 5.14.0 570

vivekpx opened the pull request:
<!--
  Make sure to have done the following:
  [] Signed off your work as per the DCO.
  [] Add unit-tests
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes # PWX-44553

**Special notes for your reviewer**:

```
[root@ip-10-38-110-203 btrfs-progs]# make test 
  TEST     fsck-tests.sh
    [TEST/fsck]   001-bad-file-extent-bytenr
    [TEST/fsck]   002-bad-transid
    [TEST/fsck]   003-shift-offsets
    [TEST/fsck]   004-no-dir-index
    [TEST/fsck]   005-bad-item-offset
    [TEST/fsck]   006-bad-root-items
    [TEST/fsck]   007-bad-offset-snapshots
    [TEST/fsck]   008-bad-dir-index-name
    [TEST/fsck]   009-no-dir-item-or-index
    [TEST/fsck]   010-no-rootdir-inode-item
    [TEST/fsck]   011-no-inode-item
    [TEST/fsck]   012-leaf-corruption
    [TEST/fsck]   013-extent-tree-rebuild
    [TEST/fsck]   014-no-extent-info
    [TEST/fsck]   016-wrong-inode-nbytes
    [TEST/fsck]   017-missing-all-file-extent
    [TEST/fsck]   018-leaf-crossing-stripes
    [TEST/fsck]   019-non-skinny-false-alert
    [TEST/fsck]   020-extent-ref-cases
    [TEST/fsck]   021-partially-dropped-snapshot-case
    [TEST/fsck]   022-qgroup-rescan-halfway
    [TEST/fsck]   023-qgroup-stack-overflow
    [TEST/fsck]   024-clear-space-cache
    [TEST/fsck]   025-file-extents
    [TEST/fsck]   026-bad-dir-item-name
    [TEST/fsck]   027-bad-extent-inline-ref-type
    [TEST/fsck]   028-unaligned-super-dev-sizes
    [TEST/fsck]   029-valid-orphan-item
    [TEST/fsck]   030-reflinked-prealloc-extents
    [TEST/fsck]   031-metadatadump-check-data-csum
    [TEST/fsck]   032-corrupted-qgroup
    [TEST/fsck]   033-lowmem-collission-dir-items
    [TEST/fsck]   034-bad-inode-flags
    [TEST/fsck]   035-inline-bad-ram-bytes
    [TEST/fsck]   036-bad-dev-extents
    [TEST/fsck]   036-rescan-not-kicked-in
    [TEST/fsck]   037-freespacetree-repair
    [TEST/fsck]   038-missing-one-file-extent
    [TEST/fsck]   039-bad-inode-mode
    [TEST/fsck]   040-compressed-nodatasum
    [TEST/fsck]   041-invalid-root-generation
    [TEST/fsck]   042-half-dropped-inode
    [TEST/fsck]   043-bad-inode-generation
    [TEST/fsck]   044-invalid-extent-item-generation
    [TEST/fsck]   045-overlap-csum-item
    [TEST/fsck]   047-dev-bytes-used
    [TEST/fsck]   049-dir-hard-link
    [TEST/fsck]   050-invalid-block-group-used
    [TEST/fsck]   051-invalid-super-bytes-used
    [TEST/fsck]   052-init-csum-tree
    [TEST/fsck]   053-bad-metadata-level
    [TEST/fsck]   054-orphan-directory
    [TEST/fsck]   055-super-num-devs-mismatch
    [TEST/fsck]   056-raid56-false-alerts
    [TEST/fsck]   057-seed-false-alerts
    [TEST/fsck]   058-bad-free-space-tree-entry
    [TEST/fsck]   059-shrunk-device
    [TEST/fsck]   060-degraded-check
    [TEST/fsck]   061-out-of-order-inline-backref
    [TEST/fsck]   063-log-missing-csum
    [TEST/fsck]   064-deprecated-inode-cache
    [TEST/fsck]   065-valid-log-tree
  TEST     mkfs-tests.sh
    [TEST/mkfs]   001-basic-profiles
    [TEST/mkfs]   002-no-force-mixed-on-small-volume
    [TEST/mkfs]   003-mixed-with-wrong-nodesize
    [TEST/mkfs]   004-rootdir-keeps-size
    [TEST/mkfs]   005-long-device-name-for-ssd
    [TEST/mkfs]   006-partitioned-loopdev
    [TEST/mkfs]   007-mix-nodesize-sectorsize
    [TEST/mkfs]   008-sectorsize-nodesize-combination
    [TEST/mkfs]   009-special-files-for-rootdir
    [TEST/mkfs]   010-minimal-size
    [TEST/mkfs]   011-rootdir-create-file
    [TEST/mkfs]   012-rootdir-no-shrink
    [TEST/mkfs]   013-reserved-1M-for-single
    [TEST/mkfs]   014-rootdir-inline-extent
    [TEST/mkfs]   015-fstree-uuid-otime
    [TEST/mkfs]   016-rootdir-bad-symbolic-link
    [TEST/mkfs]   017-small-backing-size-thin-provision-device
    [TEST/mkfs]   018-multidevice-overflow
    [TEST/mkfs]   019-basic-checksums-mkfs
    [TEST/mkfs]   020-basic-checksums-mount
    [TEST/mkfs]   021-rfeatures-quota-rootdir
    [TEST/mkfs]   022-rootdir-size
    [TEST/mkfs]   023-free-space-tree
    [TEST/mkfs]   024-fst-bitmaps
    [TEST/mkfs]   025-zoned-parallel
    [NOTRUN] skip test
    [TEST/mkfs]   026-extent-tree-to-bgt
    [TEST/mkfs]   027-rootdir-inode
    [TEST/mkfs]   028-block-group-tree
    [TEST/mkfs]   029-raid-stripe-tree
    [NOTRUN] This test requires experimental build
    [TEST/mkfs]   030-zoned-rst
    [NOTRUN] This test requires experimental build
    [TEST/mkfs]   031-zoned-bgt
    [NOTRUN] skip test
    [TEST/mkfs]   032-zoned-reset
    [NOTRUN] skip test
    [TEST/mkfs]   033-zoned-reject-rst
    [NOTRUN] skip test
    [TEST/mkfs]   034-rootdir-extra-hard-links
    [TEST/mkfs]   035-rootdir-cross-mount
    [TEST/mkfs]   036-rootdir-subvol
    [TEST/mkfs]   037-basic-compression
    [TEST/mkfs]   038-inode-flags
  TEST     misc-tests.sh
    [TEST/misc]   001-btrfstune-features
    [TEST/misc]   002-uuid-rewrite
    [TEST/misc]   003-zero-log
    [TEST/misc]   004-shrink-fs
    [TEST/misc]   005-convert-progress-thread-crash
    [TEST/misc]   006-image-on-missing-device
    [TEST/misc]   007-subvolume-sync
    [TEST/misc]   008-leaf-crossing-stripes
    [TEST/misc]   009-subvolume-sync-must-wait
    [TEST/misc]   010-convert-delete-ext2-subvol
    [TEST/misc]   011-delete-missing-device
    [TEST/misc]   012-find-root-no-result
    [TEST/misc]   013-subvolume-sync-crash
    [TEST/misc]   014-filesystem-label
    [TEST/misc]   015-dump-super-garbage
    [TEST/misc]   016-send-clone-src
    [TEST/misc]   017-recv-stream-malformatted
    [TEST/misc]   018-recv-end-of-stream
    [TEST/misc]   019-receive-clones-on-mounted-subvol
    [TEST/misc]   020-fix-superblock-corruption
    [TEST/misc]   021-image-multi-devices
    [TEST/misc]   022-filesystem-du-on-empty-subvol
    [TEST/misc]   023-device-usage-with-missing-device
    [TEST/misc]   024-inspect-internal-rootid
    [TEST/misc]   025-zstd-compression
    [TEST/misc]   026-image-non-printable-chars
    [TEST/misc]   027-subvol-list-deleted-toplevel
    [TEST/misc]   028-superblock-recover
    [TEST/misc]   029-send-p-different-mountpoints
    [TEST/misc]   030-missing-device-image
    [TEST/misc]   031-qgroup-parent-child-relation
    [TEST/misc]   032-bad-item-ptr
    [TEST/misc]   033-filename-length-limit
    [TEST/misc]   034-metadata-uuid
    [TEST/misc]   035-receive-common-mount-point-prefix
    [NOTRUN] fix not available
    [TEST/misc]   036-receive-dump-invalid-stream
    [TEST/misc]   037-fi-show-on-new-file
    [TEST/misc]   038-backup-root-corruption
    [TEST/misc]   039-receive-clone-from-current-subvolume
    [TEST/misc]   040-subvolume-delete-default
    [TEST/misc]   041-subvolume-delete-during-send
    [TEST/misc]   042-inspect-internal-logical-resolve
    [TEST/misc]   043-subvolume-set-default-toplevel
    [TEST/misc]   045-receive-check-mount-type
    [TEST/misc]   046-seed-multi-mount
    [TEST/misc]   047-raid5-convert
    [TEST/misc]   048-image-restore-mount
    [TEST/misc]   049-btrfstune-transid-mismatch
    [TEST/misc]   050-receive-prop-ro-to-rw
    [TEST/misc]   051-warning-rw-subvol-received-uuid
    [TEST/misc]   052-seed-dirty-log
    [TEST/misc]   053-receive-write-encoded
    [NOTRUN] kernel does not support send stream >1
    [TEST/misc]   054-receive-dump-newlines
    [TEST/misc]   055-qgroup-clear-stale
    [TEST/misc]   057-btrfstune-free-space-tree
    [NOTRUN] ACL is not compiled in
    [TEST/misc]   058-replace-start-enqueue
    [TEST/misc]   059-qgroup-show-stale-qgroup-crash
    [TEST/misc]   060-ino-cache-clean
    [TEST/misc]   061-reverse-receive
    [TEST/misc]   062-fi-show-raw-dm-all-devices
    [TEST/misc]   063-btrfstune-zoned-bgt
    [NOTRUN] skip test
    [TEST/misc]   064-csum-conversion-basic
    [NOTRUN] This test requires experimental build
    [TEST/misc]   065-csum-conversion-inject
    [NOTRUN] This test requires experimental build
    [TEST/misc]   066-image-filename
    [TEST/misc]   067-btrfstune-simple-quota
    [NOTRUN] This test requires experimental build
    [TEST/misc]   068-subvol-delete-recursive-show-child
  TEST     cli-tests.sh
    [TEST/cli]   001-btrfs
    [TEST/cli]   002-balance-full-no-filters
    [TEST/cli]   003-fi-resize-args
    [TEST/cli]   004-send-parent-multi-subvol
    [TEST/cli]   005-qgroup-show
    [TEST/cli]   006-qgroup-show-sync
    [TEST/cli]   007-check-force
    [TEST/cli]   008-subvolume-get-set-default
    [TEST/cli]   009-btrfstune
    [TEST/cli]   010-subvol-show-qgroup
    [TEST/cli]   011-defrag-recursion
    [TEST/cli]   012-fi-du-recursion
    [TEST/cli]   013-subvolume-delete-by-id
    [TEST/cli]   014-multiple-profiles-warning
    [TEST/cli]   015-defrag-compress
    [TEST/cli]   016-btrfs-fi-usage
    [TEST/cli]   017-fi-show-missing
    [TEST/cli]   019-subvolume-create-parents
    [TEST/cli]   020-dry-run
    [TEST/cli]   021-subvolume-multiple-arguments
    [TEST/cli]   022-fi-du-suffixes
    [TEST/cli]   023-device-delete-force
    [TEST/cli]   024-scrub-limit
    [TEST/cli]   025-subvolume-create-failures
  TEST     fuzz-tests.sh
    [TEST/fuzz]   001-simple-check-unmounted
    [TEST/fuzz]   002-simple-image
    [TEST/fuzz]   003-multi-check-unmounted
    [TEST/fuzz]   004-simple-dump-tree
    [TEST/fuzz]   005-simple-dump-super
    [TEST/fuzz]   006-simple-tree-stats
    [TEST/fuzz]   007-simple-super-recover
    [TEST/fuzz]   008-simple-chunk-recover
    [TEST/fuzz]   009-simple-zero-log
  PY       libbtrfsutil
test_start_sync (test_filesystem.TestFilesystem) ... ok
test_sync (test_filesystem.TestFilesystem) ... ok
test_wait_sync (test_filesystem.TestFilesystem) ... ok
test_snapshot_inherit (test_qgroup.TestQgroup) ... ok
test_subvolume_inherit (test_qgroup.TestQgroup) ... ok
test_add_group (test_qgroup.TestQgroupInherit) ... ok
test_new (test_qgroup.TestQgroupInherit) ... ok
test_create_snapshot (test_subvolume.TestSubvolume) ... ok
test_create_subvolume (test_subvolume.TestSubvolume) ... ok
test_default_subvolume (test_subvolume.TestSubvolume) ... ok
test_delete_subvolume (test_subvolume.TestSubvolume) ... ok
test_deleted_subvolumes (test_subvolume.TestSubvolume) ... ok
test_is_subvolume (test_subvolume.TestSubvolume) ... ok
test_read_only (test_subvolume.TestSubvolume) ... ok
test_subvolume_id (test_subvolume.TestSubvolume) ... ok
test_subvolume_id_error (test_subvolume.TestSubvolume) ... ok
test_subvolume_info (test_subvolume.TestSubvolume) ... ok
test_subvolume_info_unprivileged (test_subvolume.TestSubvolume) ... ok
test_subvolume_iterator (test_subvolume.TestSubvolume) ... ok
test_subvolume_iterator_fd_unprivileged (test_subvolume.TestSubvolume) ... ok
test_subvolume_iterator_race (test_subvolume.TestSubvolume) ... ok
test_subvolume_iterator_race_unprivileged (test_subvolume.TestSubvolume) ... ok
test_subvolume_iterator_unprivileged (test_subvolume.TestSubvolume) ... ok
test_subvolume_path (test_subvolume.TestSubvolume) ... ok

----------------------------------------------------------------------
Ran 24 tests in 33.321s

OK
```

sulakshm commented:
commit 3480373ebdf7625ee29bee65 is okay
commit 08e688fdb8f7e862092ae64cee20bc8b46 is okay
commit 1e408af31b4adfae7b6af521315b4bfae0dcf2b5 is okay
commit 9f50fd2e92e37441da3a1daa8e27fd0c400b6cdd+51c5cd3bafe5e1e8a678d661c43b0 is okay
commit 71f4ecdbb42ad is okay
commit 650c8a9c7d88f06c23a9794 is okay
commit 7437bb73f087e5f216f9c6603f5149d354e315af is okay
commit c34fc6f26ab86d03a2d47446f42b6cd4 is okay
commit to fix api interface is also okay
  Convert to struct mnt_idmap. multiple linux commits
commit fa2690af573df is okay

sulakshm commented:
commit ed2da92 removed support for folio_account_redirty but it is added here why?
except https://github.com/portworx/px-btrfs/pull/8/commits/086e7579e6ce466a521796100427c8ef0e1a2ada rest of changes are good. please can you explain why this commit deviated from linux referenced commit?

vivekpx commented:
> commit ed2da92 removed support for folio_account_redirty but it is added here why? except [086e757](https://github.com/portworx/px-btrfs/commit/086e7579e6ce466a521796100427c8ef0e1a2ada) rest of changes are good. please can you explain why this commit deviated from linux referenced commit?

This commit 44962ca for btrfs made some optimisations and removed the usage of `folio_account_redirty`  and later on commit ed2da92 restructured the code to remove `folio_account_redirty` since there was just one caller.
Ideally I should have just pulled commit 44962ca but there were other related commits and there were logic changes associated. Instead I just added back the definition of `folio_account_redirty` to keep it simple.
